src/rospkg/os_detect.py

```diff
@@ -717,6 +718,7 @@ def get_codename(self):
 OsDetect.register_default(OS_GENTOO, Gentoo())
 OsDetect.register_default(OS_LINARO, LsbDetect("Linaro"))
 OsDetect.register_default(OS_MINT, LsbDetect("LinuxMint"))
+OsDetect.register_default(OS_MX, FdoDetect("debian"))
```

Comment:
This seems like it will overmatch potentially any other debian based system. 

Trying a VM I found that lsb has it pretty well identified. 
```
demo@mx1:~
$ lsb_release -a
No LSB modules are available.
Distributor ID:	MX
Description:	MX 18.1 Continuum
Release:	18.1
Codename:	Continuum
demo@mx1:~
```

I would suggest using the LsbDetect base class. I think `LsbDetect('MX')` should work well.

Reply:
Hey, Tully,

I could not get it to work the way you proposed, and I ended up with the solution below:

I created a new detector class using the FdoDetect base class. The ros-infrastructure/rosdep#674 remains unchanged and compatible with the changes here. 

Regarding the docker image, unfortunately, I could not find one online. If it is vital for this commit to be merged, I could create one, I guess.

Thanks for your time, and sorry for the naive overmatching solution.

George

Reply:
A docker image isn't required, I created a VM for this. But docker would be much faster for testing. 

The LsbDetect seems to work fine for me in the VM.

```
osrf@mx:~
$ pip install git+https://github.com/gstavrinos/rospkg.git
Collecting git+https://github.com/gstavrinos/rospkg.git
  Cloning https://github.com/gstavrinos/rospkg.git to /tmp/pip-bWoC7k-build
Collecting PyYAML (from rospkg==1.1.7)
  Downloading https://files.pythonhosted.org/packages/9f/2c/9417b5c774792634834e730932745bc09a7d36754ca00acf1ccd1ac2594d/PyYAML-5.1.tar.gz (274kB)
    100% |████████████████████████████████| 276kB 1.1MB/s 
Collecting catkin_pkg (from rospkg==1.1.7)
  Downloading https://files.pythonhosted.org/packages/03/b5/a6092bc2b697110885421074d28168e0092636489fcaf22994375fe5038b/catkin_pkg-0.4.11.tar.gz (58kB)
    100% |████████████████████████████████| 61kB 8.8MB/s 
Collecting docutils (from catkin_pkg->rospkg==1.1.7)
  Downloading https://files.pythonhosted.org/packages/50/09/c53398e0005b11f7ffb27b7aa720c617aba53be4fb4f4f3f06b9b5c60f28/docutils-0.14-py2-none-any.whl (543kB)
    100% |████████████████████████████████| 552kB 2.7MB/s 
Collecting pyparsing (from catkin_pkg->rospkg==1.1.7)
  Downloading https://files.pythonhosted.org/packages/de/0a/001be530836743d8be6c2d85069f46fecf84ac6c18c7f5fb8125ee11d854/pyparsing-2.3.1-py2.py3-none-any.whl (61kB)
    100% |████████████████████████████████| 71kB 7.7MB/s 
Collecting python-dateutil (from catkin_pkg->rospkg==1.1.7)
  Downloading https://files.pythonhosted.org/packages/41/17/c62faccbfbd163c7f57f3844689e3a78bae1f403648a6afb1d0866d87fbb/python_dateutil-2.8.0-py2.py3-none-any.whl (226kB)
    100% |████████████████████████████████| 235kB 4.7MB/s 
Collecting six>=1.5 (from python-dateutil->catkin_pkg->rospkg==1.1.7)
  Downloading https://files.pythonhosted.org/packages/73/fb/00a976f728d0d1fecfe898238ce23f502a721c0ac0ecfedb80e0d88c64e9/six-1.12.0-py2.py3-none-any.whl
Building wheels for collected packages: PyYAML, catkin-pkg
  Running setup.py bdist_wheel for PyYAML ... error
  Complete output from command /usr/bin/python -u -c "import setuptools, tokenize;__file__='/tmp/pip-build-7GJEeD/PyYAML/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" bdist_wheel -d /tmp/tmpiZa_MKpip-wheel- --python-tag cp27:
  usage: -c [global_opts] cmd1 [cmd1_opts] [cmd2 [cmd2_opts] ...]
     or: -c --help [cmd1 cmd2 ...]
     or: -c --help-commands
     or: -c cmd --help
  
  error: invalid command 'bdist_wheel'
  
  ----------------------------------------
  Failed building wheel for PyYAML
  Running setup.py clean for PyYAML
  Running setup.py bdist_wheel for catkin-pkg ... error
  Complete output from command /usr/bin/python -u -c "import setuptools, tokenize;__file__='/tmp/pip-build-7GJEeD/catkin-pkg/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" bdist_wheel -d /tmp/tmptpwHHopip-wheel- --python-tag cp27:
  usage: -c [global_opts] cmd1 [cmd1_opts] [cmd2 [cmd2_opts] ...]
     or: -c --help [cmd1 cmd2 ...]
     or: -c --help-commands
     or: -c cmd --help
  
  error: invalid command 'bdist_wheel'
  
  ----------------------------------------
  Failed building wheel for catkin-pkg
  Running setup.py clean for catkin-pkg
Failed to build PyYAML catkin-pkg
Installing collected packages: PyYAML, docutils, pyparsing, six, python-dateutil, catkin-pkg, rospkg
  Running setup.py install for PyYAML ... done
  Running setup.py install for catkin-pkg ... done
  Running setup.py install for rospkg ... done
Successfully installed PyYAML-5.1 catkin-pkg-0.4.11 docutils-0.14 pyparsing-2.3.1 python-dateutil-2.8.0 rospkg-1.1.7 six-1.12.0
osrf@mx:~
$ python
Python 2.7.13 (default, Sep 26 2018, 18:42:22) 
[GCC 6.3.0 20170516] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import rospkg.os_detect as osd
>>> ld = osd.LsbDetect('MX')
>>> ld.is_os()
True
>>> ld.get_version()
'18.1'
>>> ld.get_codename()
'Continuum'
```